File: src/ros2_bridge/mappers/visualization_msgs/uv_coordinate.rs
//! Typed mapper for `visualization_msgs/msg/UVCoordinate`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn uv_coordinate_to_bus(msg: ros_env::visualization_msgs::msg::UVCoordinate) -> crate::visualization_msgs::msg::v1::UvCoordinate {
    crate::visualization_msgs::msg::v1::UvCoordinate {
        u: msg.u,
        v: msg.v,
    }
}

pub(crate) fn uv_coordinate_to_ros(bus: crate::visualization_msgs::msg::v1::UvCoordinate) -> ros_env::visualization_msgs::msg::UVCoordinate {
    ros_env::visualization_msgs::msg::UVCoordinate {
        u: bus.u,
        v: bus.v,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct VisualizationMsgsUvCoordinateMapper;

impl TypedTopicMapper for VisualizationMsgsUvCoordinateMapper {
    type Ros = ros_env::visualization_msgs::msg::UVCoordinate;
    type Bus = crate::visualization_msgs::msg::v1::UvCoordinate;

    fn type_name(&self) -> &'static str {
        "visualization_msgs/msg/UVCoordinate"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(uv_coordinate_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(uv_coordinate_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/visualization_msgs/uv_coordinate.py
"""Generated mapper for `visualization_msgs/msg/UVCoordinate`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.visualization_msgs.msg.v1 import UVCoordinate as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def uv_coordinate_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.u = msg.u
    bus.v = msg.v
    return bus


def uv_coordinate_to_ros(bus):
    from visualization_msgs.msg import UVCoordinate as RosMsg

    out = RosMsg()
    out.u = bus.u
    out.v = bus.v
    return out


class VisualizationMsgsUvCoordinateMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from visualization_msgs.msg import UVCoordinate as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return uv_coordinate_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return uv_coordinate_to_ros(bus)
